### ISSUE TITLE
[java] ClassNamingConventions uses wrong pattern for interfaces

### ISSUE THREAD

**Issue by zman0900** [pmd/pmd#4955]
**Affects PMD Version:**  7.0.0

**Description:**

When the ClassNamingConventions rule is configured with a custom naming pattern for abstract classes, it reports all interfaces as in violation of that rule.

Dumping AST for an interface shows that they all seem to be also marked as abstract.  
```xml
<ClassDeclaration Abstract='true' Annotation='false' Anonymous='false' BinaryName='sample.Example' CanonicalName='sample.Example' EffectiveVisibility='public' Enum='false' Final='false' Image='' Interface='true' Local='false' Nested='false' PackageName='sample' Record='false' RegularClass='false' RegularInterface='true' SimpleName='Example' Static='false' TopLevel='true' Visibility='public'>
```

**Code Sample demonstrating the issue:**

```java
package sample;

public interface Example {
}
```

```xml
<rule ref="category/java/codestyle.xml/ClassNamingConventions">
    <properties>
        <property name="abstractClassPattern" value="Abstract[a-zA-Z0-9]+" />
    </properties>
</rule>
```

**Steps to reproduce:**

1.  Configure rules file as above with custom pattern for abstract classes.
2.  Have an interface that does not meet that pattern.
3. Get violation like: `PMD Failure: sample.Example:3 Rule:ClassNamingConventions Priority:1 The interface name 'Example' doesn't match 'Abstract[a-zA-Z0-9]+'.`

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
This is weird… the rule explicitly checks for interfaces beforehand…

https://github.com/pmd/pmd/blob/c472fdf48cdf1cb8827b0ff344ff1225f89487fe/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/ClassNamingConventionsRule.java#L62-L67

**Comment by zman0900** [comment]
Oh, apparently it is wrong in 7.0.0 release but was already fixed on master: https://github.com/pmd/pmd/commit/fb496b9ee36f79f9dcdc31b180c7f0a2ac769401. Looks like this is a duplicate of #4881.
